nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,561 @@
+import io
+import string
+import os
+
+from collections import OrderedDict
+
+import pandas as pd
+
+from matplotlib import pyplot as plt
+from nilearn.datasets import load_mni152_template
+from nilearn.plotting import (plot_glass_brain,
+                              plot_roi,
+                              plot_stat_map,
+                              )
+from nilearn.plotting.js_plotting_utils import HTMLDocument
+
+from nistats.reporting import (plot_contrast_matrix,
+                               plot_design_matrix,
+                               get_clusters_table,
+                               )
+from nistats.thresholding import map_threshold
+
+
+html_template_root_path = os.path.dirname(__file__)
+
+
+def make_glm_report(
+        model,
+        contrasts,
+        title='auto',
+        roi_img=None,
+        bg_img='MNI 152 Template',
+        threshold=3.09,
+        alpha=0.01,
+        cluster_threshold=None,
+        height_control='fpr',
+        min_distance=8.,
+        plot_type='stat',
+        display_mode=None,
+        cut_coords=None,
+        nb_width=1600,
+        nb_height=800,
+        ):
+    """ Returns HTMLDocument object for a report which shows
+    all important aspects of a fitted GLM.
+    The object can be opened in a browser or saved to disk.
+    
+    Usage:
+        report = make_glm_report(model, contrasts)
+        report.open_in_browser()
+        report.save_as_html(destination_path)
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object
+        A fitted first or second level model object.
+        
+    contrasts: Dict[string, ndarray] , String
```

Comment:
what should contrasts contain, how does this parameter relate to the `contrast_def` of firstlevelmodel? could you add some details and a check that raises a suitable error if the provided contrasts are not valid?